ports/espressif/bindings/espnow/ESPNow.c

```diff
@@ -0,0 +1,876 @@
+//|         lmk: Optional[bytes],
+//|         channel: int = 0,
+//|         interface: int = 0,
+//|         encrypt: bool = False,
+//|     ) -> None:
+//|         """Modify peer.
+//|
+//|         :param bytes mac: The mac address of the peer.
+//|         :param bytes lmk: The Local Master Key (lmk) of the peer.
+//|         :param int channel: The peer's channel. Default: 0 ie. use the current channel.
+//|         :param int interface: The WiFi interface to use. Default: 0 ie. STA.
+//|         :param bool encrypt: Whether or not to use encryption."""
+//|         ...
+STATIC mp_obj_t espnow_mod_peer(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    espnow_obj_t *self = pos_args[0];
+    check_for_deinit(self);
+
+    _update_peer_info(n_args, pos_args, kw_args, true);
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(espnow_mod_peer_obj, 2, espnow_mod_peer);
+
+//|     def del_peer(self, mac: bytes) -> None:
+//|         """Delete peer.
+//|
+//|         :param bytes mac: The mac address of the peer."""
+//|         ...
+STATIC mp_obj_t espnow_del_peer(mp_obj_t self_in, mp_obj_t mac) {
+    espnow_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    uint8_t peer_addr[ESP_NOW_ETH_ALEN];
+    memcpy(peer_addr, _get_peer_addr(mac), ESP_NOW_ETH_ALEN);
+
+    check_esp_err(esp_now_del_peer(peer_addr));
+    _update_peer_count(self);
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(espnow_del_peer_obj, espnow_del_peer);
+
+// Convert a peer_info struct to python tuple
+// Used by espnow_get_peer() and espnow_get_peers()
+static mp_obj_t _peer_info_to_tuple(const esp_now_peer_info_t *peer) {
+    return MP_OBJ_NEW_TUPLE(
+        mp_obj_new_bytes(peer->peer_addr, MP_ARRAY_SIZE(peer->peer_addr)),
+        mp_obj_new_bytes(peer->lmk, MP_ARRAY_SIZE(peer->lmk)),
+        mp_obj_new_int(peer->channel),
+        mp_obj_new_int(peer->ifidx),
+        mp_obj_new_bool(peer->encrypt));
+}
+
+//|     def get_peer(self, mac: bytes) -> Tuple[bytes, int, int, bool]:
+//|         """Get the peer info for mac as a `tuple`.
+//|
+//|         :param bytes mac: The mac address of the peer.
+//|
+//|         :returns: A `tuple` of (mac, lmk, channel, interface, encrypt)."""
+//|         ...
+STATIC mp_obj_t espnow_get_peer(mp_obj_t self_in, mp_obj_t mac) {
+    espnow_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    esp_now_peer_info_t peer = {0};
+    memcpy(peer.peer_addr, _get_peer_addr(mac), ESP_NOW_ETH_ALEN);
+    check_esp_err(esp_now_get_peer(peer.peer_addr, &peer));
+
+    return _peer_info_to_tuple(&peer);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(espnow_get_peer_obj, espnow_get_peer);
+
+// --- Peer Related Properties ---
+
+//|     peers: Tuple[Tuple[bytes, bytes, int, int, bool], ...]
+//|     """The peer info records for all registered `ESPNow` peers. (read-only)
+//|
+//|     A `tuple` of tuples: ((mac, lmk, channel, interface, encrypt), ...)."""
```

Comment:
It'd probably be better to have a simple class hold data about peers instead of a tuple. tuples aren't really self documenting. BLE scans and MDNS finds use similar classes.